native-image-tests/local-scala/src/main/scala/com/lightbend/JacksonCheck.scala

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright (C) 2009-2024 Lightbend Inc. <https://www.lightbend.com>
+ */
+package com.lightbend
+
+import akka.actor.typed.ActorSystem
+import akka.serialization.SerializationExtension
+import akka.serialization.SerializerWithStringManifest
+import akka.serialization.jackson.JsonSerializable
+import com.fasterxml.jackson.annotation.JsonCreator
+import com.fasterxml.jackson.annotation.JsonProperty
+import com.fasterxml.jackson.annotation.JsonSubTypes
+import com.fasterxml.jackson.annotation.JsonTypeInfo
+import com.fasterxml.jackson.annotation.JsonTypeName
+import com.fasterxml.jackson.core.JsonGenerator
+import com.fasterxml.jackson.core.JsonParser
+import com.fasterxml.jackson.databind.DeserializationContext
+import com.fasterxml.jackson.databind.SerializerProvider
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize
+import com.fasterxml.jackson.databind.annotation.JsonSerialize
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer
+import com.fasterxml.jackson.databind.ser.std.StdSerializer
+
+object JacksonCheck {
+
+  // from the akka-serialization-jackson docs
+
+  // type hierarchy / ADT
+  // FIXME: Still haven't figured out why @JsonCreator cannot be used here
+  final case class Zoo(@JsonProperty("primaryAttraction") primaryAttraction: Animal) extends JsonSerializable
+
+  // Note: native image needs additional marker trait on ADT supertype here or else
+  //       AkkaJacksonSerializationFeature has no way to detect and register the subtypes for
+  //       reflective access
```

Comment:
This is really annoying, but can't see a way around except possibly doing classpath scanning, which we cannot do in a native image `Feature` so would have to be some other test infrastructure that users run to generate metadata files (a bit like the one I have in place internally in Akka).

Reply:
Feels like current state of art means that it is acceptable for users to have some trouble to solve themselves. Spring docs mention quite some caveats https://docs.spring.io/spring-boot/docs/current/reference/html/native-image.html#native-image.advanced.known-limitations

There I also noticed the link to https://github.com/oracle/graalvm-reachability-metadata
Is that something we can point at? I guess it's more related to libraries than user code, which is what you are trying to solve here.

Reply:
I did figure out a way around it today, apparently you can register more type reachability callbacks from a reachability callback, so it now looks for interface constructor parameters and registers a new callback handler to find and register concrete types. Seems to work fine from the test coverage in local-scala native image test.

Reply:
For the reachability metadata I've gone with including what applies from the metadata repo, but coverage there is very spotty, so I think for now it is better, for user experience, with our own verified reflection metadata for transitive dependencies we say we support (unless they provide it themselves in their jars like Netty, which then just works tm).